### PR TITLE
`Exam mode`: Improve loading animation for assessment buttons

### PR DESCRIPTION
**Pull request by Dominik-Weinzierl**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).

### Motivation and Context

This pull requests removes unwanted animation when evaluating quizzes or assessing unsubmitted exams or toggling second correction in exam mode #4620

### Description

- Instead of using `isLoading` in `exam-assessment-buttons.component.ts` for the three different functionalities (loading, evaluating Quizzes, assess Exercises), we now have `isEvaluatingQuizExercises` and `isAssessingUnsubmittedExams` as additional boolean.
- `isTogglingSecondCorrection`as Map in `assessment-dashboard.component.ts`

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Exam with at least 2 Exercises (one Quiz required)

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a Course
4. Create an Exam
5. Create Quiz Exercise and for example a Text Exercise
7. Open Assessment Dashboard of the Exam
8. Press "Evaluate quizzes" and/or "Assess unsubmitted exams" -> loading animations should be visible only on the pressed button
9. Press on one of the two "Enable second correction" buttons -> loading animations should be visible only on the pressed button

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by Strohgelaender — APPROVED** [review]
Tested on TS1, works as expected. Thanks for taking care of this 👍

**Review by Strohgelaender — DISMISSED** [review]
Tested again, both buttons are disabled but only the animation on the pressed button is shown.

File: src/main/webapp/app/entities/exercise.model.ts [hunk 1]
```diff
@@ -120,6 +120,7 @@ export abstract class Exercise implements BaseEntity {
 
     // helper attributes
     public secondCorrectionEnabled = false;
+    public isTogglingSecondCorrection?: boolean;
```
Comment (b-fein) [inline src/main/webapp/app/entities/exercise.model.ts:123]: With this change you are encoding UI/server state information inside the domain model. I think that should definitely be avoided if possible. Is there an alternative solution?

Comment (Dominik-Weinzierl) [reply src/main/webapp/app/entities/exercise.model.ts:123]: Good point. Yes, I agree. Sure, there is a better suitable solution :+1: (let me think about it :D)

**Review by Strohgelaender — APPROVED** [review]

**Review by cremertim — APPROVED** [review]
lgtm, works locally

**Pull request merged**